x-pack/plugins/secops/public/components/timeline/body/renderers/auditd_ended_session_details.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { EuiSpacer } from '@elastic/eui';
+import { get } from 'lodash/fp';
+import * as React from 'react';
+import { pure } from 'recompose';
+import styled from 'styled-components';
+
+import { BrowserFields } from '../../../../containers/source';
+import { Ecs } from '../../../../graphql/types';
+import { DraggableBadge } from '../../../draggables';
+
+import { PrimarySecondaryUserInfo } from './primary_secondary_user_info';
+import { SourceDest } from './source_dest_ip';
+
+import * as i18n from './translations';
+
+const Details = styled.div`
+  margin: 10px 0px 10px 10px;
+`;
+
+const TokensFlexItem = styled(EuiFlexItem)`
+  margin-left: 3px;
+`;
+
+interface Props {
+  id: string;
+  hostName: string | null | undefined;
+  userName: string | null | undefined;
+  primary: string | null | undefined;
+  secondary: string | null | undefined;
+  processExecutable: string | null | undefined;
+  processTitle: string | null | undefined;
+  workingDirectory: string | null | undefined;
+  args: string | null | undefined;
+  session: string | null | undefined;
+}
+
+export const AuditdEndedSessionLine = pure<Props>(
+  ({
+    id,
+    hostName,
+    userName,
+    primary,
+    secondary,
+    processExecutable,
+    processTitle,
+    workingDirectory,
+    args,
+    session,
+  }) => (
+    <EuiFlexGroup justifyContent="center" gutterSize="none" wrap={true}>
+      <TokensFlexItem grow={false}>{i18n.SESSION}</TokensFlexItem>
+      <TokensFlexItem grow={false}>
+        <DraggableBadge
+          id={`auditd-ended-session-${id}`}
+          field="auditd.session"
+          value={session}
+          iconType="number"
+        />
+      </TokensFlexItem>
+      <TokensFlexItem grow={false}>
+        <PrimarySecondaryUserInfo
+          id={`auditd-ended-session-${id}`}
```

Comment:
should (all) draggable IDs (in addition to `eventId`) include the field name?

Reply:
Whoops! Yea, that is going to blow up sky high. Will fix here and in the others. Thank you!

Reply:
I can change DraggableBadge to be:

```ts
export const DraggableBadge = pure<BadgeDraggableType>(
  ({ id, field, value, name, color = 'hollow', children, queryValue, iconType }) =>
    value != null ? (
      <DefaultDraggable
        id={`${id}-${field}-${value}`}
        field={field}
        name={name}
        value={value}
        queryValue={queryValue}
      >
        <Badge iconType={iconType} color={color}>
          {children ? children : value}
        </Badge>
      </DefaultDraggable>
    ) : null
);
```

And that would make user land code not forced to take some much responsibility

Reply:
Per conversations I included that above as well as renamed `id` to `eventId` as well as changed that issue in other parts of the code that made more sense being `eventId`